### PR TITLE
fix: de-duplicate error message

### PR DESCRIPTION
**Pull request by ig-ra**
de duplicate error message in connect -> SDK error conversion

### CODE REVIEW

File: sdk/internal/rpcerrors/errors.go [hunk 1]
```diff
@@ -9,33 +9,82 @@ import (
...
+	   	err1 = sdkerrors.ErrUnauthenticated
+	   // ErrUnauthenticated    = errors.New("unauthenticated")
+	   // 		return "unauthenticated"
+
+	   case connect.CodePermissionDenied:
+	   	err1 = sdkerrors.ErrUnauthorized
+	   // ErrUnauthorized       = errors.New("unauthorized")
+	   // 		return "permission_denied"
+
+	   case connect.CodeResourceExhausted:
```
Comment (efiShtain) [inline sdk/internal/rpcerrors/errors.go]: remove the comments

File: sdk/internal/rpcerrors/errors.go [hunk 2]
```diff
@@ -9,33 +9,82 @@ import (
...
+		return connectErr.Unwrap()
 	default:
-		err1 = sdkerrors.ErrRPC
+		sdkErr = sdkerrors.ErrRPC
 	}
 
-	if cerr := new(connect.Error); errors.As(err, &cerr) {
-		return fmt.Errorf("%w: %s (%v)", err1, cerr.Message(), cerr.Details())
+	// err is a connect error (checked in connect.CodeOf), so we can safely cast it
+	connErr := err.(*connect.Error)
```
Comment (efiShtain) [inline sdk/internal/rpcerrors/errors.go]: I think you removed the connect.CodeOf check so is this still valid conversion ?
also, why do we need this part if you already converted it before to sdk error ?

**Review by efiShtain — APPROVED** [review]
quick questions but looks ok to me if you're sure this doesn't break CLI or tests

File: sdk/internal/rpcerrors/errors.go (hunk 2 above)
Comment (ig-ra) [reply sdk/internal/rpcerrors/errors.go]: yes, conversion is done above in errors.As. There err is converted to connect.Error. If failed we return an original error

Comment (ig-ra) [reply sdk/internal/rpcerrors/errors.go]: ah. I see. you are right. unneeded conversion

**Pull request merged**